src/full/Agda/Syntax/Translation/ConcreteToAbstract.hs

```diff
@@ -2524,7 +2524,7 @@ whereToAbstract1 r whname whds inner = do
    void $ -- We can ignore the returned default A.ImportDirective.
     openModule TopOpenModule (Just am) (C.QName m) $
       defaultImportDir { publicOpen = Just noRange }
-  return (x, A.WhereDecls (am <$ whname) $ singleton d)
+  return (x, A.WhereDecls (am <$ whname) (Just am) $ singleton d)
```

Comment:
This introduces redundancy when `whname` is `Just _`.  Then you are storing the same thing twice.

Reply:
I was a bit reluctant to change what it means for the first field to be `Just _`, as this might introduce new bugs across the whole codebase that type-checking won't catch. But I get your point, I will try the boolean flag variant, let's hope the tests will catch the bugs it introduces!